.github/workflows/deploy-npm.template.yml

```diff
@@ -0,0 +1,54 @@
+name: Deploy to NPM
+
+on:
+  workflow_call:
+    inputs:
+      name:
+        required: true
+        type: string
+      working-directory:
+        required: true
+        type: string
+    outputs:
+      version:
+        description: "The NPM package version"
+        value: ${{ jobs.deploy.outputs.version }}
+
+jobs:
+  deploy:
+    if: ${{ startsWith(github.ref_name, inputs.name) }}
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.parse_version.outputs.VERSION }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          registry-url: https://registry.npmjs.org
+          node-version-file: .nvmrc
+          cache: yarn
+
+      - name: Install dependencies
+        run: yarn install --immutable
+
+      - name: Parse version
+        id: parse_version
+        env:
+          TAG: ${{ github.event.release.tag_name }}
+        run: echo "VERSION=${TAG##${{ inputs.name }}@}" >> $GITHUB_OUTPUT
+
+      - name: Set NPM version
+        working-directory: ${{ inputs.working-directory}}
+        run: npm version ${{ steps.parse_version.outputs.VERSION }} --allow-same-version --no-commit-hooks --no-git-tag-version
```

Comment:
Rather than requiring us to manually maintain the version in `package.json` we can just use the GitHub release tag as our source of truth, which is what we're essentially doing already by performing an [equality check](https://github.com/ht-sdks/events-sdk-js-mono/blob/narciero/consent-packages/.github/workflows/deploy-releases.yml#L41-L42) on the tag vs `package.json`.

Reply:
Are you saying that the npm deploy script will always bump/change the package.json version to match the github tag? And no-op if they already match?

IMO, the version is something before the npm publish: E.g. if we make commits in the repo that require a new release, that will mean a new CDN and a new NPM release. Both should require a bump in version. Both should require matching the package.json to the git tag. And both should match _eachother_. Version 1.2.3 of the NPM should be version 1.2.3 of the CDN. That was one reason to use the package.json as the source of the truth.

Can you be sure to account for the CDN publish before using this template for the main SDKs?

Reply:
> Are you saying that the npm deploy script will always bump/change the package.json version to match the github tag? And no-op if they already match?

The deploy action template has a step that invokes `npm version` prior to any build/publish step, which will set the package.json version to whatever we pass in, which in this case is the GitHub release tag version. This ensures the package that gets built and published matches the exact version from the GitHub release tag.

Will definitely verify it all works before updating any of the other packages but I think using `npm version` is the cleanest approach since it ensures a single source of truth and we don't have to remember to update any version stuff manually.